src/axom/primal/tests/primal_nurbs_patch.cpp

```diff
@@ -609,19 +611,25 @@ TEST(primal_nurbspatch, surface_evaluate)
 
   // Loop over the parameter space of the surface,
   //  and check that `evalaute` matches the results of the two isocurve methods
-  for(double u = 0; u <= 1; u += 0.1)
+
+  constexpr int npts = 11;
+  double u_pts[npts], v_pts[npts];
+  axom::numerics::linspace(0.0, 1.0, u_pts, npts);
+  axom::numerics::linspace(0.0, 1.0, v_pts, npts);
+
+  for(auto u : u_pts)
   {
-    for(double v = 0; v <= 1; v += 0.1)
+    for(auto v : v_pts)
```

Comment:
Ah, I didn't realize `numerics::linspace` had the same issue! If there was such an epsilon to `KnotVector`, I think clamping to the actual range is the way to go, but it isn't clear to me what that `eps` should be, maybe an optional parameter? For `BezierCurve` objects, that little tolerance isn't super important, since they smoothly extrapolate beyond `[0, 1]`, but that isn't the case for NURBS classes.

Would it be worth making a small change to that `linspace` function to just manually set `v[N-1] = x1` at the end of the loop?

Reply:
In the short term, I think modifying `linspace` so that the last entry is always `x1` makes sense.